freud/box.pyx

```diff
@@ -180,50 +180,72 @@ cdef class Box:
         """float: The box volume (area in 2D)."""
         return self.thisptr.getVolume()
 
-    def make_absolute(self, fractional_coordinates):
+    def make_absolute(self, fractional_coordinates, out=None):
         R"""Convert fractional coordinates into absolute coordinates.
 
         Args:
             fractional_coordinates (:math:`\left(3, \right)` or :math:`\left(N, 3\right)` :class:`numpy.ndarray`):
                 Fractional coordinate vector(s), between 0 and 1 within
                 parallelepipedal box.
+            out (:math:`\left(3, \right)` or :math:`\left(N, 3\right)` :class:`numpy.ndarray` or :code:`None`):
+                The array in which to place the absolute coordinates. It must be
+                of dtype `np.float32`. If ``None``, this function will return a
+                newly allocated array (Default value = None).
 
         Returns:
             :math:`\left(3, \right)` or :math:`\left(N, 3\right)` :class:`numpy.ndarray`:
-                Absolute coordinate vector(s).
+                Absolute coordinate vector(s). If ``out`` is provided, a
+                reference to it is returned.
         """  # noqa: E501
         fractions = np.asarray(fractional_coordinates).copy()
         flatten = fractions.ndim == 1
         fractions = np.atleast_2d(fractions)
         fractions = freud.util._convert_array(fractions, shape=(None, 3))
+        out = freud.util._convert_array(
+                out, shape=fractions.shape, allow_copy=False)
 
         cdef const float[:, ::1] l_points = fractions
         cdef unsigned int Np = l_points.shape[0]
```

Comment:
Not sure what our typical convention is here but const might be appropriate.
```suggestion
        cdef const unsigned int Np = l_points.shape[0]
```

Reply:
The way that Cython translates to C you cannot declare stack variables like this `const` because their declaration will happen well before they are initialized.